Classes/Domain/Service/ZoomApiAccessTokenFactory.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace CodeQ\ZoomApi\Domain\Service;
+
+use CodeQ\ZoomApi\Domain\Model\ZoomApiAccessToken;
+use CodeQ\ZoomApi\ZoomApiException;
+use GuzzleHttp\Client;
+use Neos\Flow\Annotations as Flow;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @Flow\Scope("singleton")
+ */
+class ZoomApiAccessTokenFactory
+{
+    private Client $client;
+
+    /**
+     * @Flow\InjectConfiguration
+     * @var array
+     */
+    protected array $settings;
+
+    /**
+     * @Flow\Inject
+     * @var LoggerInterface
+     */
+    protected $systemLogger;
+
+    /**
+     * @return void
+     * @throws ZoomApiException
+     */
+    public function initializeObject(): void
+    {
+        $zoomApiClientId = $this->settings['auth']['clientId'];
+        $zoomApiClientSecret = $this->settings['auth']['clientSecret'];
+        if (!$zoomApiClientId || !$zoomApiClientSecret) {
+            throw new ZoomApiException('Please set a Zoom Account ID, Client ID and Secret for CodeQ.ZoomApi to be able to authenticate.', 1695830249149);
+        }
+        $this->client = $this->buildClient($zoomApiClientId, $zoomApiClientSecret);
+    }
+
+    /**
+     * @return ZoomApiAccessToken
+     * @throws ZoomApiException
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function createFromConfiguration(): ZoomApiAccessToken
+    {
+        $zoomApiAccountId = $this->settings['auth']['accountId'];
+        if (!$zoomApiAccountId) {
+            throw new ZoomApiException('Please set a Zoom Account ID for CodeQ.ZoomApi to be able to authenticate.', 1695904285296);
+        }
+        $response = $this->client->post('oauth/token', [
+            'form_params' => [
+                'grant_type' => 'account_credentials',
+                'account_id' => $zoomApiAccountId
+            ]
+        ]);
+
+        if ($response->getStatusCode() !== 200) {
+            throw new ZoomApiException('Could not fetch Zoom access token. Please check the settings for account ID, client ID and client secret, as well as your Zoom app.', 1695040346621);
+        }
```

Comment:
This does not actually work, as you can see with the following test (added to `ZoomApiAccessTokenFactoryTest.php`):

https://github.com/code-q-web-factory/CodeQ.ZoomApi/blob/6a456ec6eaea8bf753374d9a8afbff119df6c8e6/Tests/Unit/ZoomApiAccessTokenFactoryTest.php#L50-L65

This test fails with the message `Failed asserting that exception of type "GuzzleHttp\Exception\ClientException" matches expected exception "CodeQ\ZoomApi\ZoomApiException".`

To fix this you either have to wrap the `client->post` with try/catch, or better yet do the following:

```suggestion
        $response = $this->client->post('oauth/token', [
            'http_errors' => false,
            'form_params' => [
                'grant_type' => 'account_credentials',
                'account_id' => $zoomApiAccountId
            ]
        ]);

        if ($response->getStatusCode() !== 200) {
            throw new ZoomApiException('Could not fetch Zoom access token. Please check the settings for account ID, client ID and client secret, as well as your Zoom app.', 1695040346621);
        }
```